source/funkin/play/PlayState.hx

```diff
@@ -192,6 +193,11 @@ class PlayState extends MusicBeatSubState
    */
   public var needsReset:Bool = false;
 
+  /**
+   * Map of the boppers that will reset their bop speed once the restart occurs.
+   */
+  public var resetBoppers:Map<Bopper, Int> = new Map<Bopper, Int>();
```

Comment:
This should certainly be part of `Stage.hx`, since all Boppers are either stage props or characters, and therefore are children of the stage.

You could also just reset bopping for all characters and all stage props on stage restart and not require a Map to store that data.

Reply:
Moved the bop speed reset into `Stage.hx` as you requested alongside cleaning some code inside `resetStage` to cycle through every single character in the `characters` Map.